protocol/message.go

```diff
@@ -39,10 +57,39 @@ type DirectoryProof struct {
 	AP    *m.AuthenticationPath
 	STR   *m.SignedTreeRoot
 	TB    *m.TemporaryBinding `json:",omitempty"`
-	Error ErrorCode           `json:",omitempty"`
+	Error ErrorCode
+}
+
+type DirectoryProofInEpoch struct {
+	Type  int
+	AP    *m.AuthenticationPath
+	STR   []*m.SignedTreeRoot
+	Error ErrorCode
+}
+
+type DirectoryProofs struct {
```

Comment:
I did it in ea4a0cc092ea99798d5f5ecbae6807ac38b5a505. I also added new `Limit` field into the `KeyLookupInEpochRequest`. What do you think?
/cc @arlolra @Liamsi